Sources/PostgresNIO/New/Connection State Machine/AuthenticationStateMachine.swift

```diff
@@ -54,10 +54,10 @@ struct AuthenticationStateMachine {
                     return self.setAndFireError(PSQLError(code: .authMechanismRequiresPassword))
                 }
                 self.state = .passwordAuthenticationSent
-                return .sendPassword(.md5(salt: salt), self.authContext)
+                return .sendPassword(.md5(salt: salt), self.authContext, promise: nil)
             case .plaintext:
                 self.state = .passwordAuthenticationSent
-                return .sendPassword(.cleartext, authContext)
+                return .sendPassword(.cleartext, authContext, promise: nil)
```

Comment:
called from channel read, I double checked and it seems to be fine to pass `nil` as the actual write does have a promise which it fulfills.